core/notifications/notifications.go

```diff
@@ -135,3 +236,44 @@ func (n *notificationService) List(limit int64, includeRead bool) ([]*model.Noti
 func (n *notificationService) isNotificationRead(notification *model.Notification) bool {
 	return notification.GetStatus() == model.Notification_Read || notification.GetStatus() == model.Notification_Replied
 }
+
+func (n *notificationService) loadNotificationObject(ctx context.Context) {
+	defer close(n.notificationCh)
```

Comment:
maybe call `n.indexNotifications(notificationContext)` directly here? Channel creates a point of synchronization and linearize order of calling `loadNotificationObject -> indexNotifications`